stm32-modules/heater-shaker/firmware/system/system_serial_number.h

```diff
@@ -0,0 +1,20 @@
+#ifndef __SYSTEM_SERIAL_NUMBER_H_
+#define __SYSTEM_SERIAL_NUMBER_H_
+
+#ifdef __cplusplus
+extern "C" {
+#endif  // __cplusplus
+
+#include <stdbool.h>
+#include <stddef.h>
+
+#include "stm32f3xx_hal.h"
+
+bool system_set_serial_number(uint64_t to_write);
```

Comment:
This is the core of the current limitation to at most 8 characters; we should think about whether we want to extent this, and how we might do it.